internal/api/v1/handlers/instance.go

```diff
@@ -156,3 +156,159 @@ func (h *InstanceHandler) GetInstance(c *fiber.Ctx) error {
 
 	return c.JSON(instance)
 }
+
+// GetPublicIPs returns a list of public IPs and their associated job IDs
+func (h *InstanceHandler) GetPublicIPs(c *fiber.Ctx) error {
+	fmt.Println("🔍 Getting public IPs...")
+
+	// Get pagination parameters
+	page := c.QueryInt("page", 1)
+	if page < 1 {
+		page = 1
+	}
+	limit := c.QueryInt("limit", 10)
+	if limit < 1 {
+		limit = 10
+	}
+	offset := (page - 1) * limit
```

Comment:
for the first page offset will be zero, did you test it if it works? I think we need to have a condition for the offset if it is zero set it to a default value.
Also it would be nice to have a helper func that receives the page and returns the ListOptions{} filled up. so handlers can use it.